tests/internal/test_wrapping.py

```diff
@@ -491,3 +493,198 @@ def f(a, b, c=None):
 
     assert closure(1, 2, 3) == (1, 2, 3, 42)
     assert channel == [((42,), {}), closure, ((1, 2, 3), {}), (1, 2, 3, 42)]
+
+
+NOTSET = object()
+
+
+class DummyWrappingContext(WrappingContext):
```

Comment:
trying to think of other scenarios.... a function wrapped with a decorator?

a function that is a context manager?

a function that is a generator?

in theory those shouldn't matter, but maybe easy enough to verify and codify their behavior here.

Reply:
Yep good point, we should add a few more examples in the tests 👍